### ISSUE TITLE
[GKE 1.7.6] Fluentd-GCP failing to export logs to Stackdriver

### ISSUE THREAD

**Issue by swizzlr** [GoogleCloudPlatform/fluent-plugin-google-cloud#168]
The following logs are visible from yesterday (can't retrieve any earlier ones):
```
2017-10-13 05:31:10 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-10-13 05:31:16 +0000 error_class="Google::Apis::TransmissionError" error="execution expired" plugin_id="object:3fd837e1f6f0"
  2017-10-13 05:31:10 +0000 [warn]: suppressed same stacktrace
```

The issue became apparent when a developer was unable to view logs for a given service on a staging server. On further investigation it was discovered that the node has not been sending logs since the 4th October, which is when this fluentd pod came online.

## Pod description

```
Name:           fluentd-gcp-v2.0-2n253
Namespace:      kube-system
Node:           gke-moonshot-staging-europ-dual-cpu-1-4c3164b5-z0l4/10.156.0.9
Start Time:     Wed, 04 Oct 2017 11:38:24 +0100
Labels:         controller-revision-hash=1419153066
                k8s-app=fluentd-gcp
                kubernetes.io/cluster-service=true
                pod-template-generation=2
                version=v2.0
Annotations:    kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"DaemonSet","namespace":"kube-system","name":"fluentd-gcp-v2.0","uid":"106757f1-9956-11e7-b505-42010a9c0fc7...
                scheduler.alpha.kubernetes.io/critical-pod=
Status:         Running
IP:             10.156.0.9
Created By:     DaemonSet/fluentd-gcp-v2.0
Controlled By:  DaemonSet/fluentd-gcp-v2.0
Containers:
  fluentd-gcp:
    Container ID:  docker://f3516c3f7d40e21df425bfdba32a01532f41a849699d6b543734536984052407
    Image:         gcr.io/google-containers/fluentd-gcp:2.0.8
    Image ID:      docker-pullable://gcr.io/google-containers/fluentd-gcp@sha256:99cff747870a01cee36c2d1d482a137cf623bf956ca122909b9dc4d831c37e2a
    Port:          <none>
    Command:
      /bin/sh
      -c
      /run.sh $FLUENTD_ARGS
    State:          Running
      Started:      Wed, 04 Oct 2017 11:38:31 +0100
    Ready:          True
    Restart Count:  0
    Limits:
      memory:  300Mi
    Requests:
      cpu:     100m
      memory:  200Mi
    Liveness:  exec [/bin/sh -c LIVENESS_THRESHOLD_SECONDS=${LIVENESS_THRESHOLD_SECONDS:-300}; STUCK_THRESHOLD_SECONDS=${LIVENESS_THRESHOLD_SECONDS:-900}; if [ ! -e /var/log/fluentd-buffers ]; then
  exit 1;
fi; LAST_MODIFIED_DATE=`stat /var/log/fluentd-buffers | grep Modify | sed -r "s/Modify: (.*)/\1/"`; LAST_MODIFIED_TIMESTAMP=`date -d "$LAST_MODIFIED_DATE" +%s`; if [ `date +%s` -gt `expr $LAST_MODIFIED_TIMESTAMP + $STUCK_THRESHOLD_SECONDS` ]; then
  rm -rf /var/log/fluentd-buffers;
  exit 1;
fi; if [ `date +%s` -gt `expr $LAST_MODIFIED_TIMESTAMP + $LIVENESS_THRESHOLD_SECONDS` ]; then
  exit 1;
fi;
] delay=600s timeout=1s period=60s #success=1 #failure=3
    Environment:
      FLUENTD_ARGS:  --no-supervisor -q
    Mounts:
      /etc/fluent/config.d from config-volume (rw)
      /host/lib from libsystemddir (ro)
      /var/lib/docker/containers from varlibdockercontainers (ro)
      /var/log from varlog (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-dj4xd (ro)
  prometheus-to-sd-exporter:
    Container ID:  docker://9debcbee1a894c2f53f2a0455d8091501de0b23ae2ec4247ef2200044053304f
    Image:         gcr.io/google-containers/prometheus-to-sd:v0.1.3
    Image ID:      docker-pullable://gcr.io/google-containers/prometheus-to-sd@sha256:c6aaa681e77e55aa7f7017ca55265accde313f8e2e5484ee1d0a4d89ff741c48
    Port:          <none>
    Command:
      /monitor
      --component=fluentd
      --target-port=31337
      --stackdriver-prefix=container.googleapis.com/internal/addons
      --whitelisted-metrics=stackdriver_successful_requests_count,stackdriver_failed_requests_count,stackdriver_ingested_entries_count,stackdriver_dropped_entries_count
    State:          Running
      Started:      Wed, 04 Oct 2017 11:38:38 +0100
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /etc/ssl/certs from ssl-certs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-dj4xd (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          True 
  PodScheduled   True 
Volumes:
  varlog:
    Type:  HostPath (bare host directory volume)
    Path:  /var/log
  varlibdockercontainers:
    Type:  HostPath (bare host directory volume)
    Path:  /var/lib/docker/containers
  libsystemddir:
    Type:  HostPath (bare host directory volume)
    Path:  /usr/lib64
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      fluentd-gcp-config-v1.1.1
    Optional:  false
  ssl-certs:
    Type:  HostPath (bare host directory volume)
    Path:  /etc/ssl/certs
  default-token-dj4xd:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-dj4xd
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  beta.kubernetes.io/fluentd-ds-ready=true
Tolerations:     :NoExecute
                 :NoSchedule
                 :NoExecute
                 :NoSchedule
                 node.alpha.kubernetes.io/ismaster:NoSchedule
                 node.alpha.kubernetes.io/notReady:NoExecute
                 node.alpha.kubernetes.io/unreachable:NoExecute
Events:          <none>
```

# After deleting the fluentd pod
The fluentd pod starts running out of memory and restarting with the following stack trace.

```
2017-10-13 14:16:53 +0000 [warn]: 'block' action stops input process until the buffer full is resolved. Check your pipeline this action is fit or not
2017-10-13 14:16:53 +0000 [warn]: 'block' action stops input process until the buffer full is resolved. Check your pipeline this action is fit or not
2017-10-13 14:16:58 +0000 [error]: error on output thread error="No such file or directory @ sys_fail2 - (/var/log/fluentd-buffers/kubernetes.system.buffer..b55b6e4edc5954a5b.log, /var/log/fluentd-buffers/kubernetes.system.buffer..q55b6e4edc5954a5b.log)"
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/plugin/buf_file.rb:75:in `rename'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/plugin/buf_file.rb:75:in `mv'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/plugin/buf_file.rb:192:in `enqueue'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:305:in `block (2 levels) in push'
  2017-10-13 14:16:58 +0000 [error]: /usr/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:304:in `block in push'
  2017-10-13 14:16:58 +0000 [error]: /usr/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:298:in `push'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:297:in `block in enqueue_buffer'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:296:in `each'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:296:in `enqueue_buffer'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:308:in `block in try_flush'
  2017-10-13 14:16:58 +0000 [error]: /usr/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:306:in `try_flush'
  2017-10-13 14:16:58 +0000 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:149:in `run'
```

## SSL error?
I found the following stack trace in the logs:

```
2017-10-13 05:30:11 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-10-13 05:30:16 +0000 error_class="Google::Apis::TransmissionError" error="execution expired" plugin_id="object:3fd837e1f6f0"
  2017-10-13 05:30:11 +0000 [warn]: /usr/lib/ruby/2.1.0/openssl/buffering.rb:345:in `write'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/ssl_socket.rb:80:in `<<'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:582:in `dump_file'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:508:in `dump'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:962:in `dump'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:517:in `block in query'
  2017-10-13 05:30:11 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:91:in `block in timeout'
  2017-10-13 05:30:11 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:101:in `call'
  2017-10-13 05:30:11 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:101:in `timeout'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:515:in `query'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:177:in `query'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1242:in `do_get_block'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1019:in `block in do_request'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1133:in `protect_keep_alive_disconnected'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1014:in `do_request'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:856:in `request'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:17:in `block in call'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:252:in `initialize'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:16:in `new'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:16:in `call'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:122:in `call_with_redirects'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:89:in `call'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:71:in `post'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:272:in `execute_once'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:107:in `block (2 levels) in execute'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:54:in `block in retriable'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `times'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `retriable'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:104:in `block in execute'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:54:in `block in retriable'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `times'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `retriable'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:96:in `execute'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/base_service.rb:353:in `execute_or_queue_command'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/generated/google/apis/logging_v2beta1/service.rb:108:in `write_entry_log_entries'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:752:in `block in write'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:580:in `each'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:580:in `write'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:354:in `write_chunk'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:333:in `pop'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:342:in `try_flush'
  2017-10-13 05:30:11 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:149:in `run'
2017-10-13 05:30:41 +0000 [warn]: temporarily failed to flush the buffer. next_retry=2017-10-13 05:30:46 +0000 error_class="Google::Apis::TransmissionError" error="execution expired" plugin_id="object:3fd837e1f6f0"
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/openssl/buffering.rb:326:in `syswrite'
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/openssl/buffering.rb:326:in `do_write'
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/openssl/buffering.rb:344:in `write'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/ssl_socket.rb:80:in `<<'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:582:in `dump_file'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:508:in `dump'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/http.rb:962:in `dump'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:517:in `block in query'
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:91:in `block in timeout'
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:101:in `call'
  2017-10-13 05:30:41 +0000 [warn]: /usr/lib/ruby/2.1.0/timeout.rb:101:in `timeout'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:515:in `query'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient/session.rb:177:in `query'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1242:in `do_get_block'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1019:in `block in do_request'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1133:in `protect_keep_alive_disconnected'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:1014:in `do_request'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/httpclient-2.8.3/lib/httpclient.rb:856:in `request'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:17:in `block in call'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:252:in `initialize'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:16:in `new'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_client_adapter.rb:16:in `call'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:122:in `call_with_redirects'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:89:in `call'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/hurley-0.2/lib/hurley/client.rb:71:in `post'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:272:in `execute_once'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:107:in `block (2 levels) in execute'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:54:in `block in retriable'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `times'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `retriable'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:104:in `block in execute'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:54:in `block in retriable'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `times'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/retriable-2.1.0/lib/retriable.rb:48:in `retriable'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/http_command.rb:96:in `execute'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/lib/google/apis/core/base_service.rb:353:in `execute_or_queue_command'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/google-api-client-0.9.28/generated/google/apis/logging_v2beta1/service.rb:108:in `write_entry_log_entries'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:752:in `block in write'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:580:in `each'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluent-plugin-google-cloud-0.6.4/lib/fluent/plugin/out_google_cloud.rb:580:in `write'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:354:in `write_chunk'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/buffer.rb:333:in `pop'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:342:in `try_flush'
  2017-10-13 05:30:41 +0000 [warn]: /var/lib/gems/2.1.0/gems/fluentd-0.12.37/lib/fluent/output.rb:149:in `run'
```

**Comment by swizzlr** [comment]
I have very little familiarity with the guts of fluentd, but I've now drained and am deleting the node. Hopefully the issue won't recur, but I was quite surprised to see there was no health check reporting that fluentd was failing to drain its buffer.

**Comment by igorpeshansky** [comment]
@crassirostris, any ideas?

**Comment by crassirostris** [comment]
>  I was quite surprised to see there was no health check reporting that fluentd was failing to drain its buffer

There is a healthcheck, if you look at the pod definition. If the pod is failing to send logs for a prolonged amount of time, healthcheck assumes that the problem is in buffers and removes them

I looked up your cluster and it seems you had timeout errors, which is possible under specific load profile. There are two outputs and it seems that only one of them was broken, probably that's why liveness probe was failing to fix the problem. I created a follow-up: https://github.com/kubernetes/kubernetes/issues/53931
